2016-2017_Starstruck/CrossBot Competition/src/CrossDriver.c

```diff
@@ -0,0 +1,302 @@
+      if (lcdReadButtons(uart1)==LCD_BTN_LEFT) {
+        count--;
+      }
+      else if (lcdReadButtons(uart1)==LCD_BTN_RIGHT) {
+        count++;
+      }
+      else if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=DRIVE_VAL;
+      }
+      break;
+		case BATTERY_MENU:
+      lcdSetText(uart1,1,"Battery  Voltage");
+      lcdSetText(uart1,2,"<    Select    >");
+      if (lcdReadButtons(uart1)==LCD_BTN_LEFT) {
+        count--;
+      }
+      else if (lcdReadButtons(uart1)==LCD_BTN_RIGHT) {
+        count++;
+      }
+      else if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=BATTERY_VAL;
+      }
+      break;
+    case SELF_MENU:
+      lcdSetText(uart1,1,"   Self Test    ");
+      lcdSetText(uart1,2,"<    Select    >");
+      if (lcdReadButtons(uart1)==LCD_BTN_LEFT) {
+        count--;
+      }
+      else if (lcdReadButtons(uart1)==LCD_BTN_RIGHT) {
+        count++;
+      }
+      else if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=SELF_VAL;
+      }
+      break;
+    case MANUAL_MENU:
+      lcdSetText(uart1,1,"  Manual Test   ");
+      lcdSetText(uart1,2,"<    Select    >");
+      if (lcdReadButtons(uart1)==LCD_BTN_LEFT) {
+        count--;
+      }
+      else if (lcdReadButtons(uart1)==LCD_BTN_RIGHT) {
+        count++;
+      }
+      else if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=MANUAL_VAL;
+      }
+      break;
+    case MEMES_MENU:
+      lcdSetText(uart1,1,"   Dank Memes   ");
+      lcdSetText(uart1,2,"<    Select    >");
+      if (lcdReadButtons(uart1)==LCD_BTN_LEFT) {
+        count--;
+      }
+      else if (lcdReadButtons(uart1)==LCD_BTN_RIGHT) {
+        count=POT_MENU;
+      }
+      else if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=MEMES_VAL;
+      }
+      break;
+    case 	POT_VAL: //potentiometer readings
+      lcdPrint(uart1,1,"Drive: %d", analogRead(RP));
+      if(lcdReadButtons(uart1)==LCD_BTN_CENTER)
+        count = POT_MENU; //go back to select menu
+      break;
+    case DRIVE_VAL: //get current drive mode
+      lcdPrint(uart1,1,"Drive Mode:");
+      lcdPrint(uart1,2,"Function Disabled");
+
+      if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=DRIVE_MENU;
+      }
+      break;
+		case BATTERY_VAL: //get battery voltage
+		  lcdPrint(uart1, 1, "Main: %dmV",powerLevelMain()); //Display main battery on LCD
+		  lcdPrint(uart1, 2, "Expander: %dmV",(int)(analogRead(EXPANDER)*35.84)); //Display power expander battery on LCD
+      if(lcdReadButtons(uart1)==LCD_BTN_CENTER) {
+        count=BATTERY_MENU;
+      }
+      break;
+    case SELF_VAL: //self test
+      //currentConfig = holonomicDrive;
+      lcdClear(uart1);
+      lcdPrint(uart1, 2, "Don't Touch!!!!!");
+      for(unsigned char x=1;x<=10;x++) {
+        lcdPrint(uart1, 1, "Motor %d",x);
+        pulseMotor(x);
+      }
+			delay(50);
+      count=SELF_MENU;
+      break;
+    case MANUAL_VAL: //manual single motor test
+      lcdClear(uart1);
```

Comment:
i might be able to put the manualCount integer just in here i dont remember if i use it anywhere else but in MANUAL_VAL so it would make more sense to just keep it here instead of outside